internal/mage/engine.go

```diff
@@ -308,13 +318,42 @@ func (t Engine) Dev(ctx context.Context) error {
 	return nil
 }
 
+const cniVersion = "v1.2.0"
+
+func dnsnameBinary(c *dagger.Client, arch string) *dagger.File {
+	src := c.Git("https://github.com/vito/dnsname", dagger.GitOpts{KeepGitDir: true}).
+		Branch("fork").
+		Tree()
+
+	return c.Container(dagger.ContainerOpts{
```

Comment:
nit: if possible would be good to use `util.GoBase` so we pick up the common image version, cache mounts, etc.